File: Runtime/TransitionAdjacencySettingMap.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'************************************************************************************

Class TransitionAdjacencySettingMap

    Private m_Map As New Dictionary(Of Integer, TransitionAdjacencySetting)

    Public Sub New(ByVal settings As TransitionAdjacencySettingCollection)

        For Each s As TransitionAdjacencySetting In settings
            Me.AddItem(s)
        Next

    End Sub

    Public Sub AddItem(ByVal setting As TransitionAdjacencySetting)

        If (Not Me.m_Map.ContainsKey(setting.TransitionGroupId)) Then
            Me.m_Map.Add(setting.TransitionGroupId, setting)
        End If

    End Sub

    Public Function GetItem(ByVal transitionGroupId As Integer) As TransitionAdjacencySetting

        If (Me.m_Map.ContainsKey(transitionGroupId)) Then
            Return Me.m_Map(transitionGroupId)
        Else
            Return Nothing
        End If

    End Function

End Class

File: src/Runtime/TransitionAdjacencySettingMap.cs
// stsim: A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2024 Apex Resource Management Solutions Ltd. (ApexRMS). All rights reserved.

using System.Collections.Generic;

namespace SyncroSim.STSim
{
    internal class TransitionAdjacencySettingMap
    {
        private Dictionary<int, TransitionAdjacencySetting> m_Map = new Dictionary<int, TransitionAdjacencySetting>();

        public TransitionAdjacencySettingMap(TransitionAdjacencySettingCollection settings)
        {
            foreach (TransitionAdjacencySetting s in settings)
            {
                this.AddItem(s);
            }
        }

        public void AddItem(TransitionAdjacencySetting setting)
        {
            if (!this.m_Map.ContainsKey(setting.TransitionGroupId))
            {
                this.m_Map.Add(setting.TransitionGroupId, setting);
            }
        }

        public TransitionAdjacencySetting GetItem(int transitionGroupId)
        {
            if (this.m_Map.ContainsKey(transitionGroupId))
            {
                return this.m_Map[transitionGroupId];
            }
            else
            {
                return null;
            }
        }
    }
}
